tests/undo/test_trash_gc.py

```
@@ -0,0 +1,1301 @@
+        from undo.trash_gc import TrashDeleteResult, TrashGC
+
+        trash = tmp_path / "trash"
+        trash.mkdir()
+        target = trash / "no_rename_perm"
+        target.mkdir()
+        (target / "x").write_text("x")
+        gc = TrashGC(trash, journal_path=tmp_path / "j")
+
+        sentinel_exc = OSError(13, "simulated permission denied")
+
+        def failing_rename(src: object, dst: object) -> None:
```

Ruff (0.15.11)

[warning] Unused function argument: `src`

(ARG001)